### DISCUSSION TITLE
make docker-whl  (build error: incompatible engine node version)

### DISCUSSION THREAD

**Discussion by sanyaade-teachings** [learningequality/kolibri#12547] [Community]
Hi Everyone,
I started to work around kolibri docker  build tools. I was able to build the base image, the demoserver built but failed due to missing pex file which I could build with docker-whl  I got the following error while running the make docker-whl :

![image](https://github.com/user-attachments/assets/923ffafc-4c10-4e2a-9d84-cdfe7095880b)

The above error is due to the fact that the build expected [ engine "node" is incompatible with this module. Expected version "18.x". Got "12.22.9" ] . 

I went through both the docker/build-whl.dockerfile [setup for nodejs 18.19.0]; 
![image](https://github.com/user-attachments/assets/4307afcf-54e5-4bdc-bafe-4d2c41c3ff74)

package.json also has it setup for 18.19.0
![image](https://github.com/user-attachments/assets/ca5ede75-d514-4270-8155-3ca5bc5a4d44)

I could not find the configuration file that sets it for nodejs 12.22.9  ( this is my black box as I am studying and navigating kolibri ).
I will be grate for any help, pointer or direction. Many thanks!

Hear from you all soon.

God blesses!!!

Regards

**Comment by bjester** [comment]
I'm not positive this is the issue but I believe the line `ENV NODE_VERSION=18.19.0` is actually incorrect, and I'm surprised we haven't run into this before. That should be a build argument based off how it's used. 

So some ideas for how to resolve this:
1) Could you change `ENV` to `ARG` on that previously described line in the docker file and see if that resolves your issue?
2) Could you try clearing the `yarn_cache` docker volume? i.e. `docker volume rm yarn_cache`

**Reply by sanyaade-teachings** [reply]
> I'm not positive this is the issue but I believe the line `ENV NODE_VERSION=18.19.0` is actually incorrect, and I'm surprised we haven't run into this before. That should be a build argument based off how it's used.
> 
> So some ideas for how to resolve this:
> 
> 1. Could you change `ENV` to `ARG` on that previously described line in the docker file and see if that resolves your issue?
> 2. Could you try clearing the `yarn_cache` docker volume? i.e. `docker volume rm yarn_cache`

@bjester 👍 
Many thanks for your support. I followed all your directions and steps but still the same error. All other docker images built successfully though, only the docker-whl giving this error message, 
I may have to play around with the manual build step and then re-create this docker-whl. 

I need to have kolibri in container on my self-hosted portainer platform.

Many thanks for your help and support.

God blesses!!!

Regards

**Comment by rtibbles** [comment]
I think the main issue here is that we don't use the `make docker-whl` command for anything any more, as it was previously used in our buildkite workflow, which we have stopped using, and so we haven't made any updates to it as a consequence. I think it would probably be better to clean up the unused buildkite related files, as they will continue to go unmaintained.

**Reply by sanyaade-teachings** [reply]
> I think the main issue here is that we don't use the `make docker-whl` command for anything any more, as it was previously used in our buildkite workflow, which we have stopped using, and so we haven't made any updates to it as a consequence. I think it would probably be better to clean up the unused buildkite related files, as they will continue to go unmaintained.

@rtibbles : I am interested in the Docker for kolibri. I will go through the developer documentation and replicate them for docker environment. I was also thinking about docker-compose way of running kolibri, studio, etc...

Also I have come accross comments in some of the source files that stated that some parameter e.g. kolibri ports are hard coded. I went through the env.list, and others, just to pin down where the build is picking up ==> e.g "got 12.22.4", even looked/skim through inside yarn.lock.

I run many applications both for research and teachings hence the dockerized and microservices are the way to go.

Many thanks for your support. Much appreciated.

God blesses!!!

Regards

**Comment by MisRob** [comment]
Closing as it seems resolved as far as I can say
